### ISSUE TITLE
"--include-merged-tags" not working in "lerna publish" and "lerna version"

### ISSUE THREAD

**Issue by xueran** [lerna-lite/lerna-lite#507]
### Describe the bug

when use "lerna changed",it is found 1 package ready to publish, but when use  "lerna publish" or  "erna version", it is assuming all packages changed

### Expectation

 In the above questions，found 1 package ready to publish is right, so expect the "lerna publish" and "lerna version" behavior to be consistent with the "lerna changed" behavior

### Reproduction

Git log need contain "Merge branch xxx".

The following steps may be required for problem recurrence:
1. Create a new branch
2. publish a package,
3. merge the branch into the master,
4. create another branch,
5. and change another package for publishing

### Lerna config and logs

<details><summary>lerna.json</summary><p>

```json
<!-- Please paste your `lerna.json` here -->
```
</p></details>

<details><summary>lerna-debug.log</summary><p>

```txt
<!-- If you have a `lerna-debug.log` available, please paste it here -->
<!-- Otherwise, feel free to delete this <details> block -->
```
</p></details>

### Environment Info

  System:
    OS: macOS 11.3.1
    CPU: (12) x64 Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
    Memory: 126.13 MB / 16.00 GB
    Shell: 5.8 - /bin/zsh
 
 Binaries:
    Node: 16.13.0 - ~/.nvm/versions/node/v16.13.0/bin/node
    Yarn: 1.22.19 - ~/.nvm/versions/node/v16.13.0/bin/yarn
    npm: 8.1.0 - ~/.nvm/versions/node/v16.13.0/bin/npm
  
Browsers:
    Chrome: 110.0.5481.177
    Firefox: 78.9.0
    Safari: 14.1

### Used Package Manager

pnpm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/lerna-lite/lerna-lite/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [docs](https://github.com/lerna-lite/lerna-lite#readme).
- [X] Check that there isn't [already an issue](https://github.com/lerna-lite/lerna-lite/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/lerna-lite/lerna-lite/discussions).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by ghiscoding** [comment]
can't help without a repro and this code came from the original Lerna code

**Comment by xueran** [comment]
> can't help without a repro that I can clone, also this code came from the original Lerna code and the behavior didn't change. I would also suggest to look at Lerna's [issues](https://github.com/lerna/lerna/issues) to see if there's anything similar. It's also not very clear if run this command from master branch or a different branch, Lerna doesn't always produce expected result when on a different branch, for example this Stack Overflow [question](https://stackoverflow.com/questions/55522919/bump-version-and-publish-packages-from-one-branch-but-keep-tags-in-another-branc)
> 
> A possible alternative is to use `lerna publish from-package`

I also use Lerna, and  the result of Lerna is right. Currently, there is no way to provide repo, because this is the company's private library.

**Comment by ghiscoding** [comment]
The implementation should be the same as Lerna, there aren't much differences in code from to Lerna vs Lerna-Lite (the only major difference is that Nrwl are starting to add a lot of Nx related code which will not be added to Lerna-Lite). 

There are Jest unit tests that exist for `lerna version --include-merged-tags` in [here](https://github.com/lerna-lite/lerna-lite/blob/main/packages/version/src/__tests__/version-include-merged-tags.spec.ts) and these tests are still passing. The code came from Lerna, I don't think that I modified anything related to it, so it should work as intended

**Comment by xueran** [comment]
> The implementation should be the same as Lerna, there aren't much differences in code from to Lerna vs Lerna-Lite (the only major difference is that Nrwl are starting to add a lot of Nx related code which will not be added to Lerna-Lite).
> 
> There are Jest unit tests that exist for `lerna version --include-merged-tags` in [here](https://github.com/lerna-lite/lerna-lite/blob/main/packages/version/src/__tests__/version-include-merged-tags.spec.ts) and these tests are still passing. The code came from Lerna, I don't think that I modified anything related to it, so it should work as intended


The reason is found. It is the tag pattern problem. The lerna-lite `publish` and `version` command will pass the `isIndependent` parameter when calling `collectUpdates`, and set `tagPattern="*@*"` in `collectUpdates`. The code is as follows:

https://github.com/lerna-lite/lerna-lite/blob/4fe2a7051f41052665b5da601eabf3fed803a746/packages/core/src/utils/collect-updates/collect-updates.ts#L47

In version `conventional-commits`, tagPattern is also set：

https://github.com/lerna-lite/lerna-lite/blob/4fe2a7051f41052665b5da601eabf3fed803a746/packages/version/src/conventional-commits/get-commits-since-last-release.ts#L57


For some reasons, my project cannot set tagPattern as `*@*`, so whether it can support custom tagPattern.Thank you very much.

**Comment by ghiscoding** [comment]
in that case it must be related to issue #220 and PR #311, @StarpTech might have more details to share, in any case feel free to create a Pull Request if you know how to fix it

**Comment by xueran** [comment]
My usage scenario is in the company's automated process, so the tag is automatically added by the automated process, and the module name is used as the tag, such as `lerna-project_ 1-0-32-1_ PD_ BL`.

**Comment by ghiscoding** [comment]
@xueran 
By the way, on a totally different subject, I saw that you opened a new issue in Lerna not long ago

> The publishConfig of lerna is different from pnpm , it is not support "main" , "module" and "types"

a similar issue was also opened in Lerna-Lite under #404 and we did resolve this issue with PR #415, so this should work as expected in Lerna-Lite. I've set a list of properties to be overridden, you can see the list below (the list is actually a copy of what pnpm have in their own repo)... so anyway, I just thought of letting you know in case you didn't know that this issue was actually fixed in Lerna-Lite (requires >=1.13.0) but wasn't fixed in Lerna itself (at least not yet)

https://github.com/lerna-lite/lerna-lite/blob/2e9dc7aef1d98f006d78b1596af2acdc37266f4c/packages/publish/src/lib/override-publish-config.ts#L3-L21

**Comment by StarpTech** [comment]
@ghiscoding I don't know the repo so well. Filtering tags based on `'*@*'` was required not to include tags from others that aren't managed by lerna lite. I don't see how it can conflict with lerna at all. Based on your provided information, it's hard to understand your use case @xueran. 

> I also use Lerna, and the result of Lerna is right. Currently, there is no way to provide repo, because this is the company's private library.

Please provide a repro with your lerna.json, and expectations. The source is not important.
